client/types.go

```diff
@@ -12,9 +12,11 @@ const providerKey = "provider_key"
 
 // ApiResponse - formatted response to client
 type ApiResponse struct {
-	Reason     string
-	Success    bool
-	StatusCode int
+	Reason         string
+	Success        bool
+	StatusCode     int
+	limitRemaining *int
+	limitReset     *int
```

Comment:
Why are these pointers? You seldom see pointers to (machine) word-sized integers because it is more efficient to just store them in the same space a pointer would consume, plus you save the indirection caused by a second access to memory.

Reply:
Because the zero value for integers is 0 and I  did not want it to be confusing. If someone calls the function and the extension is not being used, prefer to return nil than zero, wdyt?

Reply:
I think then I'd prefer to have a `LimitRemaining` struct with both ints so that you'd only need to check one pointer for `nil`, and if it exists, then you can read the ints without further indirection nor checks.

Reply:
Ok so you mean to wrap the limits in some struct, maybe `limitExtensions` or something and assume its safe to do a single check for nil?

Reply:
yep, if the headers was parsed (even if the `Atoi` failed, which you can deal with separately) you then fill the two integers in this struct and set a pointer to it in the response (otherwise it just stays with `nil`).